### ISSUE TITLE
Save and Save & publish

### ISSUE THREAD

**Issue by Tychodewaard** [dnnsoftware/Dnn.Platform#3009]
## Description of problem
There are quite a few customers that want to edit content and save but not yet publish. For various reasons like: 
- it's time to go home, tomorrow I'll finish
- I think it's good but I want my colleague to co read
Currently, when you save, the new content is publlished immediately.

## Description of solution
Other systems have 2 buttons: 
1. Save
2. Save 7 publish

## Additional context
For tenders this is a must have for most occasions. So, if we want DNN to have a go at government institutions we really need to have this.

**Comment by valadas** [comment]
This feature currently exists in the Html module, in the module settings you have that option for "Direct Publish" and "Content Staging". As for other modules, then it is up to the individual module to implement something similar. There are other issues open to bring more advanced workflows to the platform too. For that reason I am closing this issue for now but if I misunderstood something, feel free to comment.

**Comment by Tychodewaard** [comment]
In that case I don't understand the workflow.
If I set the workflow on module level on 'Content staging', edit content and click 'save' I see my work directly. Alos when I am not logged in.
If I verify my settings, I see the workflow is back on Direct publish.

Should I do this in another way? I don't see the workflow on site settings for instance...

**Comment by valadas** [comment]
Hmm, that is not supposed to happen, let me give it a try here...

**Comment by valadas** [comment]
Hmm, interesting, the Content staging workflow does not save for the module level, it looks like it saves for the page level though, reopening this issues for that reason.

**Comment by Tychodewaard** [comment]
Is there a setting for page level? I didn't know that...Can you point out where?
Because if it doesn't exist, the workflow or 'Save and Save & publish' only exists in the database.

**Comment by valadas** [comment]
Yes, it is in each html module settings, you can change the workflow (Direct Publish or Content Staging for platform, Evoq I believe has more options and the underlying/data logic supports adding more although there is no UI for creating them). Then right under that you can choose if that applies to the module, the page or the site and you can override existing settings.

![image](https://user-images.githubusercontent.com/6371568/66163916-6d055d00-e5ff-11e9-8695-dbf54069abd0.png)

**Comment by valadas** [comment]
Now I am not sure where this needs to be fixed, here or in https://github.com/DNN-Connect/CKEditorProvider

Basically I am not sure where the module stops and where the provider starts, might be interesting to test another provider to check if it's a problem with the provider or module.

**Comment by Tychodewaard** [comment]
This should not be in the CKeditor as that would make it depending on the editor. And that is not what you want. So, it should on module level (taking advantage of generic logic) IMHO

**Comment by valadas** [comment]
Cool, so taking it as a bug under this repository then.

**Comment by hismightiness** [comment]
I just ran into this today in a demo.  :D

**Comment by WillStrohl** [comment]
Oh man... and I just ran into this again today attempting to answer [a forum post](https://dnncommunity.org/forums/aft/244)!  😆 Whoops!
